lib/rules/no-tabs.js

```diff
@@ -21,25 +21,45 @@ module.exports = {
             category: "Stylistic Issues",
             recommended: false
         },
-        schema: []
+        schema: [],
+        fixable: "whitespace"
     },
 
     create(context) {
         return {
             Program(node) {
+                let totalLength = 0;
+
                 context.getSourceLines().forEach((line, index) => {
-                    const match = regex.exec(line);
+                    let match;
+
+                    // Intentional use of conditional assignment - .exec() returns null when it
+                    // cannot find more matches, breaking the loop
+                    // eslint-disable-next-line no-cond-assign
+                    while (match = regex.exec(line)) {
+                        const loc = {
+                            line: index + 1,
+                            column: match.index + 1
+                        };
 
-                    if (match) {
-                        context.report(
+                        context.report({
                             node,
-                            {
-                                line: index + 1,
-                                column: match.index + 1
-                            },
-                            "Unexpected tab character."
-                        );
+                            loc,
+                            message: "Unexpected tab character.",
+
+                            // This function needs to capture the current `match`
+                            // eslint-disable-next-line no-loop-func
```

Comment:
Regarding unexpected token for template: Make sure you set `parserOptions:
{ ecmaVersion: 6 }` in the test case object.

On Aug 27, 2016 1:25 AM, "Robert Rossmann" <notifications@github.com> wrote:

> In lib/rules/no-tabs.js
> <https://github.com/eslint/eslint/pull/6987#discussion_r76511470>:
>
> >                              node,
> > -                            {
> > -                                line: index + 1,
> > -                                column: match.index + 1
> > -                            },
> > -                            "Unexpected tab character."
> > -                        );
> > +                            loc,
> > +                            message: "Unexpected tab character.",
> > +
> > +                            // This function needs to capture the current `match`
> > +                            // eslint-disable-next-line no-loop-func
>
> I believe this was a problem with var due to them being scoped to the
> function, not to current block. However, this rule uses let and const
> which are block-scoped and thus the fix function will see the intended
> value when called. Nevertheless, I have added tests for this.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/eslint/eslint/pull/6987/files/12749f7ca3560148f1149b875a1643547d42c60f#r76511470>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AARWegiK6yYyrO4HnvCW4Q-of404rj79ks5qj9hwgaJpZM4JuijX>
> .
>